docs/design/pnc.md

```diff
@@ -0,0 +1,192 @@
+      - id: '12355'
+        target: prod/rhdm-7.10.0.redhat-00004-update.zip
+```
+
+These artifacts are fetched to `artifacts/<target>` path at the root of the repository.
+
+### PNC instance configuration
+Currently, the url of the PNC instance is configured as a part of OSBS config. 
+
+
+
+[atomic_reactor/plugins/fetch_maven_artifacts.py](https://github.com/containerbuildsystem/atomic-reactor/blob/master/atomic_reactor/plugins/fetch_maven_artifacts.py)
+1) OSBS loads the `fetch-artifacts-pnc.yaml`
+2) Information about individual artifacts is fetched from PNC API at `/artifacts/{id}` endpoint ([docs](https://github.com/project-ncl/pnc-api/blob/5b35ad6fe22769510f60b0025752dc6c954c9734/src/main/java/org/jboss/pnc/api/repositorydriver/dto/RepositoryArtifact.java))
+   - `publicUrl` key is used to get download url
+   - supported checksums are saved
+   - build ids are saved
+3) Download queue is formed containing above info for each artifact
+4) SBOM component info for each artifact is fetched from PNC @ `/artifacts/{id}`, `purl` key
+5) Queue is executed and each artifact is downloaded, checksum verified
+6) SBOM is compiled, including PNC components from step 4)
+
+
+### Missing features
+
+PNC SBOM components don't have the most complete purl, e.g.:
+`pkg:maven/org.kie.trustyai/explainability-service@999.0.0.managedsvc-redhat-00001?type=zip`
+
+According to [purl spec](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#maven), `classfier` and
+`type` parameters should be also defined, method is currently pending depending on if the artifacts are single components,
+or products that contain other components. Purl should also specify a `repository_url` parameter.
+
+
+
+## Design for the implementation in Cachi2
+
+### PNC config
+Existing `fetch-artifacts-pnc.yaml` file ([docs](https://osbs.readthedocs.io/en/osbs_ocp3/users.html?highlight=fetch#fetch-artifacts-pnc-yaml))
+will be used as configuration. It needs to be extended to also include PNC instance url.
+
+### Pre-fetch PNC dependencies
+Cachi2 will use the `artifacts` key specified in the `fetch-artifacts-pnc.yaml` file to fetch info about given artifacts
+from PNC (`/artifacts/{id}` endpoint), including url to download the artifact, checksums to verify integrity, and the purl.
+The artifacts should be fetched into a separate subdirectory in cachi2 output folder, and into the target path.
+
+#### Fetching sources
+Cachi2 will use the `build_id` keys to fetch build sources from PNC (`/builds/{id}/scm-archive` endpoint). These sources
+should be fetched into a separate subdirectory in cachi2 output folder, different from the artifacts.
+
+### Generating the SBOM
+As mentioned in [missing features](#missing-features), the purl needs to specify more parameters
+- `repository_url` - obtainable by using host part of `publicUrl` and the path in `targetRepository.repositoryPath`
+  from PNC `/artifacts/{id}` endpoint response.
+- `type` & `classifier` - solution pending
```

Comment:
What is blocking us from figuring this point out? Can the solution be released without these two qualifiers?

Reply:
We're currently discussing what's the best way to determine those. The easy way is to look at the file extension, or a deeper inspection. It does not block the solution I think, since we still have time before we have to implement this part.